cmd/system-shutdown/system-shutdown-utils.h

```diff
@@ -29,4 +29,8 @@ void die(const char *msg);
 __attribute__ ((format(printf, 1, 2)))
 void kmsg(const char *fmt, ...);
 
+// Reads a possible argument for reboot syscall in /run/systemd/reboot-param,
+// which is the place where systemd stores it.
+int read_reboot_arg(char *arg, int max_size);
```

Comment:
Re: size_t I always have mixed feelings wrt using natural integer size vs array sizes, it is a C language issue in the end. In fact, fgets uses "int", but in this case the advantage of using size_t is that you make sure that max_size is non-negative, so I will change the type.